### PR TITLE
NAS-102183 / 11.3 / fix(jail/common_validation): Validate jail names as well

### PR DESCRIPTION
**Pull request by skarekrow**
NAS-102183

Signed-off-by: Brandon Schneider <brandon@ixsystems.com>

### CODE REVIEW

File: src/middlewared/middlewared/plugins/jail.py [hunk 1]
```diff
@@ -343,6 +344,14 @@ def common_validation(self, verrors, options, update=False, jail=None):
                         'Please Enter two valid and different '
                         f'space/comma-delimited MAC addresses for {key}.'
                     )
+        if options['uuid']:
+            valid = True if re.match(r"^[a-zA-Z0-9\._-]+$", name) else False
+
+            if not valid:
+                verrors.add(
+                    'options.uuid',
+                    'Invalid character in {options["uuid"]}, please remove it.
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/jail.py]: F strings ;)

Comment (william-gr) [reply src/middlewared/middlewared/plugins/jail.py]: What is the invalid character? Or it should say what are the valid characters, perhaps?

File: src/middlewared/middlewared/plugins/jail.py [hunk 2]
```diff
@@ -343,6 +344,18 @@ def common_validation(self, verrors, options, update=False, jail=None):
...
+        if options['uuid']:
+            valid = True if re.match(
+                r"^[a-zA-Z0-9\._-]+$", options['uuid']
+            ) else False
+
+            if not valid:
+                verrors.add(
+                    'options.uuid',
+                    f'Invalid character in {options["uuid"]}. '
+                    'Valid options are alphanumeric characters, '
```
Comment (Mrt134) [inline src/middlewared/middlewared/plugins/jail.py]:
```suggestion
                    'Alphanumeric, period (.), underscore (_), '
```

File: src/middlewared/middlewared/plugins/jail.py [hunk 3]
```diff
@@ -343,6 +344,18 @@ def common_validation(self, verrors, options, update=False, jail=None):
...
+                    'a period, underscore or dash.'
```
Comment (Mrt134) [inline src/middlewared/middlewared/plugins/jail.py]:
```suggestion
                    'and dash (-) characters are allowed.'
```

File: src/middlewared/middlewared/plugins/jail.py [hunk 4]
```diff
@@ -343,6 +344,18 @@ def common_validation(self, verrors, options, update=False, jail=None):
                         'Please Enter two valid and different '
                         f'space/comma-delimited MAC addresses for {key}.'
                     )
+        if options['uuid']:
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/jail.py]: @skarekrow i am not sure this key will exist on update

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/jail.py]: Can you please verify ?

**Pull request merged**